### PR TITLE
MHV-55401 Fix allergies error on medications list page load

### PR DESCRIPTION
**Pull request by gerard-sullivan**
## Summary

- Allergies error modal doesn't show up when we are not downloading something.
- Allergies error modal shows up when we print the page on medications and medication detail page.

## Related issue(s)

- https://jira.devops.va.gov/browse/MHV-55401

## Testing done

- Tested locally

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

Prescriptions List page and Prescription Detail page

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by jamigibbs — DISMISSED** [review]

File: src/applications/mhv/medications/containers/Prescriptions.jsx [hunk 1]
```diff
@@ -308,11 +309,20 @@ const Prescriptions = () => {
...
           );
+        } else {
+          updateLoadingStatus(false, '');
         }
+      } else if (
+        prescriptionsFullList?.length &&
+        allergiesError &&
+        pdfTxtGenerateStatus.status === PDF_TXT_GENERATE_STATUS.InProgress
+      ) {
+        updateLoadingStatus(false, '');
```
Comment (vmenshutin-bylight) [inline src/applications/mhv/medications/containers/Prescriptions.jsx:320]: I think this if else condition could be simplified (i.e. to not write `updateLoadingStatus(false, '');` twice

File: src/applications/mhv/medications/containers/PrescriptionDetails.jsx [hunk 2]
```diff
@@ -275,11 +275,28 @@ const PrescriptionDetails = () => {
...
+      if (
+        allergies &&
+        pdfTxtGenerateStatus.status === PDF_TXT_GENERATE_STATUS.NotStarted &&
+        pdfTxtGenerateStatus.format === 'print'
+      ) {
+        window.print();
+      }
+    },
+    [allergies, pdfTxtGenerateStatus],
+  );
```
Comment (vmenshutin-bylight) [inline src/applications/mhv/medications/containers/PrescriptionDetails.jsx:299]: can this be moved into the previous `useEffect`?

File: src/applications/mhv/medications/containers/PrescriptionDetails.jsx [hunk 3]
```diff
@@ -349,6 +366,9 @@ const PrescriptionDetails = () => {
       generatePDF();
     } else if (pdfTxtGenerateStatus.format === DOWNLOAD_FORMAT.TXT) {
       generateTXT(buildAllergiesTXT());
+    } else {
+      setPdfTxtGenerateStatus({ status: PDF_TXT_GENERATE_STATUS.NotStarted });
```
Comment (vmenshutin-bylight) [inline src/applications/mhv/medications/containers/PrescriptionDetails.jsx:370]: do we need to set status here? i.e. two previous conditions are not doing it.

File: src/applications/mhv/medications/containers/Prescriptions.jsx [hunk 4]
```diff
@@ -321,8 +331,30 @@ const Prescriptions = () => {
     ],
   );
 
+  useEffect(
```
Comment (vmenshutin-bylight) [inline src/applications/mhv/medications/containers/Prescriptions.jsx:334]: can this `useEffect` be combined with the one above?

**Review by vmenshutin-bylight — CHANGES_REQUESTED** [review]

File: src/applications/mhv/medications/containers/PrescriptionDetails.jsx (hunk 3 above)
Comment (gerard-sullivan) [reply src/applications/mhv/medications/containers/PrescriptionDetails.jsx:370]: @vmenshutin-bylight Yeah, we should do it. The above conditions are doing it in `generatePDF` and `generateTXT` function.

**Review by vmenshutin-bylight — DISMISSED** [review]

**Review by vmenshutin-bylight — APPROVED** [review]

**Review by jamigibbs — APPROVED** [review]

**Pull request merged**